client/components/preview/SideBar.js

```diff
@@ -1,8 +1,9 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable react/no-direct-mutation-state */
```

Comment:
no-direct-mutation-state is a big one. That's an error for ES lint, and it's why we can't commit from here. When updating React state, we're supposed to use the setState method.